File: bindings/python/python/maxt/examples/account_and_safety.py
"""Read an Upbit account when configured and build safe request objects."""

from __future__ import annotations

import asyncio
import os
from decimal import Decimal

from maxt import (
    Client,
    Exchange,
    Market,
    Network,
    OrderRequest,
    Side,
    Size,
    TransferHistoryRequest,
    UpbitAdapter,
)


async def main() -> None:
    market = Market.spot(Exchange.UPBIT, "BTC", "KRW")
    draft = OrderRequest.limit_order(
        market,
        Side.BUY,
        Size.base(Decimal("0.0001")),
        Decimal("100000"),
        client_id="docs-example-only",
    )
    history = TransferHistoryRequest(asset="BTC", network=Network.BITCOIN, limit=20)
    print(f"order draft only; it was not sent: {draft}")
    print(f"transfer-history request: {history}")

    access_key = os.getenv("UPBIT_ACCESS_KEY")
    secret_key = os.getenv("UPBIT_SECRET_KEY")
    if not access_key or not secret_key:
        print("set UPBIT_ACCESS_KEY and UPBIT_SECRET_KEY to run the read-only account section")
        return

    client = Client(UpbitAdapter(access_key=access_key, secret_key=secret_key))
    balances = await client.balances()
    orders = await client.open_orders()
    print(f"{len(balances)} balances and {len(orders)} open orders")


if __name__ == "__main__":
    asyncio.run(main())

File: bindings/dart/example/account_and_safety.dart
import 'package:maxt/maxt.dart';

const _accessKey = String.fromEnvironment('UPBIT_ACCESS_KEY');
const _secretKey = String.fromEnvironment('UPBIT_SECRET_KEY');

Future<void> main() async {
  final market = Market.spot(Exchange.upbit, 'BTC', 'KRW');
  final draft = OrderRequest.limit(
    market,
    Side.buy,
    Size.base(Decimal.parse('0.0001')),
    Decimal.parse('100000'),
  ).withClientId('docs-example-only');
  final history = TransferHistoryRequest(
    asset: 'BTC',
    network: Network.bitcoin,
    limit: 20,
  );
  print('Order draft only; it was not sent: ${draft.clientId}');
  print('Transfer-history request: ${history.asset}');

  if (_accessKey.isEmpty || _secretKey.isEmpty) {
    print(
      'Pass -DUPBIT_ACCESS_KEY=... -DUPBIT_SECRET_KEY=... for read-only account data.',
    );
    return;
  }

  await Maxt.initialize();
  try {
    final client = Client(
      UpbitAdapter(accessKey: _accessKey, secretKey: _secretKey),
    );
    final balances = await client.balances();
    final orders = await client.openOrders();
    print('${balances.length} balances and ${orders.length} open orders');
  } finally {
    await Maxt.dispose();
  }
}
